### PR TITLE
Clean up SerialCommander for use in `exec`.

### PR DESCRIPTION
**Pull request by andrewjstone**
There was a problem where every drain operation waited until its timeout making it take longer than necessary. Additionally, some of the parsing was less than ideal.

Instead we now pass through the unedited output from the terminal minus the command we sent, and the prompt. In other words we strip the first and last lines of the output. This gives us nice terminal colors for our output as well.

We may want to also set the terminal size for better sized output as well right after we login. That can come in a follow up.

### CODE REVIEW

File: lib/src/serial.rs [hunk 1]
```diff
@@ -40,12 +44,20 @@ impl SerialCommander {
         name: String,
         log: Logger,
     ) -> SerialCommander {
+        let prompt_regex =
+            //Regex::new(&format!("root@(?:{name}|unknown): # ")).unwrap();
```
Comment (andrewjstone) [inline lib/src/serial.rs:48]: Before we set the hostname, we get 'unknown'. I tried and failed to make the regex specific. It probably doesn't matter.

**Comment by rcgoodfellow** [comment]
Looks like there may be some issues with the `debian-11` image we use in `a4x2`

```
exec: [sc] ce: timeout waiting for data
Mar 14 08:36:33.452 ERRO launch failed: exec: [sc] ce: timeout waiting for data
```

**Comment by andrewjstone** [comment]
> Looks like there may be some issues with the `debian-11` image we use in `a4x2`
> 
> ```
> exec: [sc] ce: timeout waiting for data
> Mar 14 08:36:33.452 ERRO launch failed: exec: [sc] ce: timeout waiting for data
> ```

One problem here was due to the mechanism to look for the login prompt. I was only taking helios into account and being overly specific. However, now I also think I may have messed up the actual regular prompt too because things are stalling out and hanging. I think I need to restore your `PROMPT_COMMAND` that I removed rather than using a regex to detect the command prompt. That's much less flakey, and unlikely to conflict with with other strings, which I assume is why you put it in there in the first place. However, in order to make the rest of the code work like it does now, we'll have to wait for the next line to get the actual prompt after we detect the `PROMPT_COMMAND`. This isn't much of an issue.

Unfortunately, I gotta run now. I'll do this later tonight or over the weekend.

**Review by rcgoodfellow — APPROVED** [review]
Thanks @andrewjstone! Took this for a spin on a4x2 and did some testing with commands we use from CI. Things look good

**Pull request merged**